Cython/Utility/TypeConversion.c

```diff
@@ -742,7 +740,7 @@ static CYTHON_INLINE PyObject* {{TO_PY_FUNCTION}}({{TYPE}} value, Py_ssize_t wid
     length = end - dpos;
     ulength = length;
     prepend_sign = 0;
-    if (!is_unsigned && value <= neg_one) {
+    if (!is_unsigned && value < 0) {
```

Comment:
IIRC, we had this at some point, and it gives a warning for unsigned types, because they are never `<0`.